Tables migration

Zasa-san opened the pull request:
fixes #7052 #6431

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

Zasa-san committed:
migrate user table

Zasa-san committed:
migrate translations table

Zasa-san committed:
Merge branch 'development' into 7052-migrate-tables

Zasa-san committed:
migrate pagelist table

Zasa-san committed:
migrate language list

Zasa-san committed:
migrate custom uploads

Zasa-san committed:
migrated activity log table

Zasa-san committed:
Merge branch 'development' into 7052-migrate-tables

Zasa-san committed:
Merge branch '7052-migrate-tables' of https://github.com/huridocs/uwazi into 7052-migrate-tables

Zasa-san committed:
update menu table + refactor + test

Zasa-san committed:
migrate relationships list table

Zasa-san committed:
add default sorting state to log

Zasa-san committed:
lint fix

Zasa-san committed:
Merge branch 'development' into 7052-migrate-tables

Zasa-san committed:
updated menu blocker

Zasa-san committed:
updated pages e2e

Zasa-san committed:
refactor + more tests

Zasa-san committed:
permissions table migrated

Zasa-san committed:
Merge branch 'development' into 7052-migrate-tables

Zasa-san committed:
filters table migration + empty groups fix

Zasa-san committed:
typo fix

mfacar commented on app/react/V2/Routes/Settings/Filters/components/helpers.ts:
```diff
@@ -60,65 +64,69 @@ const updateFilters = (
+
+    if (subRows) {
+      sanitizedFilter.items = subRows.map(item => {
+        const { rowId: itemRowId, _id, ...sanitizedItem } = { ...item };
```
The destructuration of { ...item } seems not to be needed, actually { rowId: itemRowId, _id, ...sanitizedItem } could be directly in the params of the map like subRows.map(({rowId: itemRowId, _id, ...sanitizedItem }) => ...)

mfacar commented on app/react/V2/Routes/Settings/Filters/FiltersTable.tsx:
```diff
@@ -70,43 +71,57 @@ const FiltersTable = () => {
     setSettings(response);
     setDisabled(false);
     setHasChanges(false);
+    const formattedFilters = formatFilters(savedFilters || []);
```
Could it be that the reported inconsistencies are related with keeping the local filters instead of the data in the response?

Zasa-san replied:
Kinda yes. It was not a good idea to avoid using the revalidator.

mfacar commented on app/react/V2/Routes/Settings/Filters/components/FiltersSidepanel.tsx:
```diff
@@ -60,19 +60,21 @@ const FiltersSidepanel = ({
 
     if (!filter?._id) delete result._id;
     if (!filter?.id) result.id = uniqueID();
+    if (result.rowId === defaultValues.rowId) result.rowId = uniqueID();
```
To avoid this comparison, uniqueID() could be assigned directly instead of the value 'NEW'

Zasa-san replied:
Calling the function for uniqueID on the instantiation of the form was triggering a rerender loop. I had a hard time figuring out was was triggering the endless rerenders on the component and removing the uniqueID from the default values fixed it.

mfacar replied:
I got it, it doesn't matter 👍

mfacar reviewed (changes requested):
Great job! I did some tests having the following comments:
* The Filters page has some unexpected behaviors after doing some editions:
      - Editing an instance without filters, adding several filters, saving, deleting 1 filter, the table is emptied, the data is correct after reloading.
      - After some editions, DnD starts working wrongly, some elements can't be moved out or into a group, or moving an item into a group ends with moving other items into it or changing the order. Two times when this  happened, select all starts a loop rendering having the error "Maximum update depth exceeded"
* Some actions column headers don't have the bottom border. 
![Screenshot 2024-08-28 191331](https://github.com/user-attachments/assets/375ef417-6ffc-4b27-a961-7f1d7aebdf4f)
* Filters and Menu items are not translatable.
* It isn't clear to me what is the solution for #6431

There are also a couple of things from previous developments that maybe can be easily fixed in this PR.
- Filters action buttons have a different order than other pages
![Screenshot 2024-08-28 192426](https://github.com/user-attachments/assets/5ff1d5ac-af66-4ed6-9803-7a1569721e5a)
- Relationship Types show the translated value followed by the original name wrapped by parenthesis, this is different in other pages like templates 
![Screenshot 2024-08-28 192927](https://github.com/user-attachments/assets/7abc8244-0a52-4a49-9665-f615237b6b40)

Zasa-san committed:
Merge branch 'development' into 7052-migrate-tables

Zasa-san committed:
fix filters behavior

Zasa-san committed:
remove translate for consistency

Zasa-san committed:
remove unneeded translate

Zasa-san commented:
> Great job! I did some tests having the following comments:
> 
>     * The Filters page has some unexpected behaviors after doing some editions:
>       - Editing an instance without filters, adding several filters, saving, deleting 1 filter, the table is emptied, the data is correct after reloading.
>       - After some editions, DnD starts working wrongly, some elements can't be moved out or into a group, or moving an item into a group ends with moving other items into it or changing the order. Two times when this  happened, select all starts a loop rendering having the error "Maximum update depth exceeded"

This was a mistake in rowIds, fixed!

> 
>     * Some actions column headers don't have the bottom border.
This is a problem on all these new tables, if it's simple I'll fix it in this PR, otherwise it's going to be fixed separately.
> 
>     * Filters and Menu items are not translatable.

Menu items are translatable after saving. Can you confirm that? Translate them before saving would not be possible.
 
Filters it's a whole other problem. Groups are translatable after saving, they are part of the `Filters` context, but single filters are not. They are the actual template names and do not exist on the `Filters` context. I could make them translatable but the user would have no way of knowing that what they are translating is the NAME of the template, not the name of the filter. I will remove translate from the filter UI for items, since I don't think it's a good idea that users translate tamplate names by mistake.

>     * It isn't clear to me what is the solution for [Difficult to determine editing row on larger screens #6431](https://github.com/huridocs/uwazi/issues/6431)

Rows highlighted as you hover over them ;)
> 
> 
> There are also a couple of things from previous developments that maybe can be easily fixed in this PR.
> 
>     * Filters action buttons have a different order than other pages
Fixed!

>     * Relationship Types show the translated value followed by the original name wrapped by parenthesis, this is different in other pages like templates

That's right, removed the translate there.

Zasa-san committed:
sort buttons

Zasa-san committed:
Merge branch 'development' into 7052-migrate-tables

Zasa-san committed:
refactor .map

Zasa-san committed:
Merge branch 'development' into 7052-migrate-tables

Zasa-san committed:
apply sr-only on the content, not the cell, for actions

Zasa-san committed:
correctly handle changes and reset

Zasa-san committed:
add rowId afther rerender

mfacar reviewed (approved):
LGTM! Really important improvements! 💯

Zasa-san committed:
typo fixes

Zasa-san committed:
Merge branch 'development' into 7052-migrate-tables

Zasa-san committed:
removed previous table and unused dnd component

Zasa-san committed:
update translations

Zasa-san committed:
Merge branch 'development' into 7052-migrate-tables

aphilop commented:
Some questions/comments to check if they should be part of this PR:

- I am wondering why in the Filters table we don't show "Selected x of y" next to the Delete button when item(s) are selected?

<img width="1524" alt="image" src="https://github.com/user-attachments/assets/48bfe3eb-b3cc-471b-9c54-8738c422c50a">

- Same as above in the Uploads table. Also in this table the action buttons at the bottom are not at the left as in all other tables.

![image](https://github.com/user-attachments/assets/0e8bf445-91ad-4115-b94b-b63918e9f22f)

Zasa-san commented:
> Some questions/comments to check if they should be part of this PR:
> 
> * I am wondering why in the Filters table we don't show "Selected x of y" next to the Delete button when item(s) are selected?
> 
> 
> 
> * Same as above in the Uploads table. Also in this table the action buttons at the bottom are not at the left as in all other tables.
> 

@aphilop We have a few UI inconsistencies in the settings screen that are the result of iterating over the designs as we develop each screen.

I would suggest having an issue to list these inconsistencies, and to work on it after we finish all the settings screens (I think only templates is missing now). 

We have this issue here to get started: https://github.com/huridocs/uwazi/issues/6424

And then some minor issues that could be solved on a final pass on the settings UI:

https://github.com/huridocs/uwazi/issues/2032
https://github.com/huridocs/uwazi/issues/6837
https://github.com/huridocs/uwazi/issues/6757
https://github.com/huridocs/uwazi/issues/6425
https://github.com/huridocs/uwazi/issues/6435

Usually we don't find time to address these minor issues, so it could be a good thing to compile them into a big one and solve them after we have migrated all the settings screens.

Zasa-san committed:
Merge branch 'development' into 7052-migrate-tables

Zasa-san committed:
Merge branch 'development' into 7052-migrate-tables
